pubsub/lib/pubsub_samples_subscriber.ex

```diff
@@ -0,0 +1,56 @@
+defmodule GoogleApi.PubSub.Samples.Subscriber do
+  use Task
+  @moduledoc """
+  Subscriber sample for GoogleApi.PubSub
+  """
+
+  # def init(state), do: {:ok, state}
+
+  def start_link(project_id, subscription_name) do
+    task = Task.async(GoogleApi.PubSub.Samples.Subscriber, :listen, [project_id, subscription_name])
+    {:ok, task.pid}
+  end
+
+  @doc """
+  Publish a message to a PubSub topic.
+
+  ## Examples
+
+      iex> GoogleApi.PubSub.Samples.publish("YOUR_PROJECT_ID", "test-topic", "This is a message")
+      "published message 159505488737289"
+
+  """
+  def listen(project_id, subscription_name) do
+    {:ok, token} = Goth.Token.for_scope("https://www.googleapis.com/auth/cloud-platform")
+    conn = GoogleApi.PubSub.V1.Connection.new(token.token)
+
+    # Make a subscription pull
+    {:ok, response} = GoogleApi.PubSub.V1.Api.Projects.pubsub_projects_subscriptions_pull(
+      conn,
+      project_id,
+      subscription_name,
+      [body: %GoogleApi.PubSub.V1.Model.PullRequest{
+        maxMessages: 10
+      }]
+    )
+
+    if response.receivedMessages != nil do
+      for message <- response.receivedMessages do
```

Comment:
Better to use `Enum.each` here. `for` is a generator in Elixir, not a looping construct. This generates an list (full of `:ok` atoms) that you don't need.